### PR TITLE
refactor(menu): deprecate submenu block

### PR DESCRIPTION
**Pull request by sjohnson-sage**
BREAKING CHANGE: The SubmenuBlock component has been deprecated.
It is no longer required to build a submenu.
If you need a scrollable submenu, please use ScrollableBlock

### Proposed behaviour
Remove the `SubmenuBlock` component and update any tests etc to maintain 100% coverage

### Current behaviour
`SubmenuBlock` component exists in Carbon

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Give all the `Menu` stories a once over http://localhost:9001/?path=/docs/design-system-menu--default-story

### CODE REVIEW

File: src/components/menu/__internal__/submenu/submenu.component.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ const Submenu = React.forwardRef(
     const menuContextOpen = menuContext.openSubmenu;
 
     const formattedChildren = React.Children.map(children, (child) => {
-      if (child.type === ScrollableBlock) {
+      if (child && child.type === ScrollableBlock) {
```
Comment (dawidzarzycki) [inline src/components/menu/__internal__/submenu/submenu.component.js:46]: I guess this change not corresponding to the PR title and I guess it should be as a different commit at least in this case

Comment (sjohnson-sage) [reply src/components/menu/__internal__/submenu/submenu.component.js:46]: It can't be actually because removing the check for `SubmenuBlock` in `MenuItem` causes this to error, so the commit would be broken if I split it out

Comment (sjohnson-sage) [reply src/components/menu/__internal__/submenu/submenu.component.js:46]: More specifically removing the `React.Children.map` bit on line 45 means that null children can now (correctly) be passed into `Submenu` and this extra check is needed. It should have been there anyway

**Review by dawidzarzycki — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
